platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/AddressBookInitializer.java

```diff
@@ -151,6 +161,12 @@ public AddressBook getInitialAddressBook() {
      */
     @NonNull
     private AddressBook initialize() {
+        final int versionComparison =
+                loadedSoftwareVersion == null ? 1 : currentVersion.compareTo(loadedSoftwareVersion);
+        if (versionComparison < 0) {
+            throw new IllegalStateException("The current software version `" + currentVersion + "` is prior to the software version `"
```

Comment:
seems like this check has nothing to do with the AB. maybe it belongs somewhere else

Reply:
I have moved the version comparison into the Browser, right after loading the saved state from disk.